functions/placeShipsRandom.ts

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { Ship } from "../interfaces/types";
+
+// x=col y=row
+// Add a thing to check for coordinate outside the grid
+const placeShipsRandom = (
+  ship: Ship,
+  Board: number[][],
+  setBoard: React.Dispatch<React.SetStateAction<number[][]>>
+) => {
+  const orientation: string =
+    Math.round(Math.random()) >= 0.5 ? "horizontal" : "vertical";
+  // "horizontal";
+
+  const updatedBoard: number[][] = [...Board];
+  // Since array extends from 0 to 9 in row and column
+  const x: number = Math.floor(Math.random() * 10);
+  const y: number = Math.floor(Math.random() * 10);
+
+  const isOutsideGrid = (x: number, y: number, orientation: string) => {
+    if (orientation === "horizontal" && x + ship["type"] > 10) {
```

Comment:
ship is an `Object`. Don't access it like an array unless the key is dynamic. In this case, it's fixed as `type`, so it could be accessed as `ship.type`. This is much easier to write and shows up on autocomplete.